Add to-many relationship format as an array of link objects

andrewsardone opened the pull request:
Initially [proposed](https://github.com/json-api/json-api/issues/183#issuecomment-33840493) by @mikelinington in json-api/json-api#183, this expands the to-many relationship format to accomodate an array of `link` objects.

``` json
{
  "posts": [
    {
      "id": "1",
      "title": "One Type Purr Author",
      "links": {
        "authors": [
          {
            "href": "http://example.com/people/9",
            "id": "9",
            "type": "people"
          },
          {
            "href": "http://example.com/cats/1",
            "id": "1",
            "type": "cats"
          }
        ]
      }
    }
  ]
}
```

andrewsardone committed:
Flip order of to-one & to-many relationship sections

I think it makes sense to start by explaining to-one relationships,
followed by to-many relationships. The simpler case of a to-one
relationship helps build up the format in the reader's mind.

andrewsardone committed:
Add to-many format proposed by @mikelinington

Initially proposed in [a comment][ac] within json-api/json-api#183, this
expands the to-many relationship format to accomodate an array of `link`
objects.

[ac]: https://github.com/json-api/json-api/issues/183#issuecomment-33840493

dgeb commented:
Reviewing issues this morning, I was reminded that @bencates [originally proposed](https://github.com/json-api/json-api/issues/164#issuecomment-31991224) this in #164. 

Regardless of origin, I think it's a good idea that rounds out the possible methods for specifying links, and it does so in a consistent format.

We should probably define "link" objects higher up in the "Relationships" section and expand the possibilities to mention an array of "link" objects. For example:

```
The link to each related resource MUST be one of the following:

* a string or number - to represent a single ID.
* an array of strings or numbers - to represent multiple IDs.
* a "link" object that contains one or more of the attributes:
  `"id"`, `"ids"`, `"href"` and `"type"`. Note that `"id"` and `"ids"` should
  never be present together.
* an array of "link" objects
```

Since the term "link" won't actually appear in a JSON document, I believe it should be written "link" instead of `"link"`.

Anyway, these are minor nits. First things first: I'm :+1: on this change.

andrewsardone committed:
Define "link" objects in Relationships section

Per @dgeb's [suggestion][s].

[s]: https://github.com/json-api/json-api/pull/187#issuecomment-33905900

andrewsardone committed:
Remove code-literal wrapping of "link" objects

Per @dgeb's [comment][c], these objects won't appear in the JSON under a
literal `"link"` key, so they shouldn't written as such.

[c]: https://github.com/json-api/json-api/pull/187#issuecomment-33905900

andrewsardone commented:
Good notes, @dgeb – thanks! I pushed [two commits](https://github.com/json-api/json-api/compare/andrewsardone:8f475c1...andrewsardone:ee6ce8a) with the suggested updates.

dgeb commented:
@andrewsardone looks good to me - thanks!

andrewsardone commented:
Do we have anything major remaining here, or is this good to merge?

dgeb commented:
I'm :+1: on this - just waiting for feedback from @steveklabnik and @wycats

steveklabnik commented:
Makes sense to me. @wycats ?

dgeb commented:
Just chatted with @wycats, who gave his ok to merge.

Thanks @andrewsardone :shipit:
